itou/users/models.py

```diff
@@ -1125,6 +1125,21 @@ class Meta:
                     "Un utilisateur ayant un NIR ne peut avoir un motif justifiant l'absence de son NIR."
                 ),
             ),
+            # We can't traverse relation in CHECK constraint so we need to use an hardcoded PK
+            models.CheckConstraint(
+                condition=(
+                    Q(birth_country=None, birth_place=None)
+                    | Q(birth_country__isnull=False, birth_country_id=Country.FRANCE_ID, birth_place__isnull=False)
+                    | Q(
+                        ~Q(birth_country__isnull=False, birth_country_id=Country.FRANCE_ID),
+                        Q(birth_place__isnull=True),
+                    )
+                ),
+                name="jobseekerprofile_birth_country_and_place",
+                violation_error_message=(
+                    "La commune de naissance ne doit être spécifiée que quand le pays de naissance est la France"
```

Comment:
```suggestion
                    "La commune de naissance ne doit être spécifiée que quand le pays de naissance est la France."
```
(les autres messages ont des points finaux)